blog/2025-05-24-inspektor-gadget-inspect-cmd.md

```diff
@@ -0,0 +1,405 @@
+		Content:     []byte(repository),
+	}
+	ociInfo.Data["oci.tag"] = &api.GadgetInspectAddendum{
+		ContentType: "text/plain",
+		Content:     []byte(tag),
+	}
+	ociInfo.Data["oci.digest"] = &api.GadgetInspectAddendum{
+		ContentType: "text/plain",
+		Content:     []byte(digest),
+	}
+	ociInfo.Data["oci.created"] = &api.GadgetInspectAddendum{
+		ContentType: "text/plain",
+		Content:     []byte(created),
+	}
+
+	gadgetCtx.SetVar("extraInfo.oci", ociInfo)
+
+	return nil
+}
+```
+
+#### Serializing and Loading "Extra" Info in `GetGadgetInfo`
+
+Once the "extra" info is stored in the `gadgetctx`, the final task is to serialize it — and in the case of gRPC clients, potentially load it back into the client-side `gadgetctx`.
+
+```go
+// pkg/gadget-context/gadget-context.go
+
+func (c *GadgetContext) SerializeGadgetInfo(extraInfo bool) (*api.GadgetInfo, error) {
+  ...
+  if c.ExtraInfo() && extraInfo {
+      gi.ExtraInfo = &api.ExtraInfo{
+          Data: make(map[string]*api.GadgetInspectAddendum),
+      }
+
+      for k, v := range c.GetVars() {
+          if !strings.HasPrefix(k, "extraInfo.") {
+              continue
+          }
+          for key, val := range v.(*api.ExtraInfo).Data {
+              gi.ExtraInfo.Data[strings.TrimPrefix(key, "extraInfo.")] = val
+          }
+      }
+  }
+  return gi, nil
+}
+
+
+func (c *GadgetContext) LoadGadgetInfo(info *api.GadgetInfo, paramValues api.ParamValues, run bool, extraInfo *api.ExtraInfo) error {
+  ...
+  if c.ExtraInfo() && extraInfo != nil {
+      for k, v := range extraInfo.Data {
+          // Key is in the form of "wasm.upcalls", "ebpf.sections", etc.
+          prefix := strings.Split(k, ".")[0]
+          ei, ok := c.GetVar("extraInfo." + prefix)
+          if !ok {
+              ei = &api.ExtraInfo{
+                  Data: make(map[string]*api.GadgetInspectAddendum),
+              }
+              c.SetVar("extraInfo."+prefix, ei)
+          }
+          ei.(*api.ExtraInfo).Data[k] = v
+      }
+  }
+  return nil
+}
+```
+
+### UX Enhancements & Demo
+
+There's a lot of useful data our command can show, but dumping it all at once isn't exactly user-friendly. To make things smoother and easier to navigate, we've added some handy feature flags that give you more control over what you see and how you see it:
+
+```bash
+  -o, --output         string   Output mode: json, jsonpretty, yaml (default "jsonpretty")
+  --extra-info         string   Specify the particular information to retrieve
+  --jsonpath           string   JSONPath expression to extract from the extra info
+  --show-datasources   bool     Display datasources along with their available fields
+```
+
+In this section, we walk through a demonstration of these features. Since the raw output can be verbose, we utilize `jq` and our utility flags to cleanly extract and display meaningful insights.
+
+---
+
+**1. Extracting all available info keys using `jq`:**
+
+```bash
+$ sudo ig image inspect fsnotify:main | jq keys
```

Comment:
Don't specify any tag so that it points to `latest`:

```suggestion
$ sudo ig image inspect fsnotify | jq keys
```

Same in other places below.